pyqtgraph/icons/__init__.py

```diff
@@ -12,14 +12,20 @@ class GraphIcon:
 
     def __init__(self, path):
         self._path = path
-        self._icon = None
         name = path.split('.')[0]
```

Comment:
I know this line isn't part of this commit, but this line I suspect should be changed; or at least put in the docstring what the appropriate type for `path` should be.  Perhaps we should use `os.path.basename()` instead?

Reply:
I put a better explanation

Reply:
My apologies for not sharing specific concerns  The use-case I'm imagining is someone tries to use an image/icon that is not part of the library, and they need to provide a path that has some directories in it.  While that would work w/ this code, the name would include the whole path before the first period.  If we want just the filename we should likely use `name = os.path.basename(path)`


In the docstring you do specify that the images must be in the library; but is that a necessary restriction?

Reply:
this should not be a user facing feature in that sense. That is why it is called `GraphIcon`